lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -0,0 +1,305 @@
+'use strict';
+
+const _ = require('lodash');
+const url = require('url');
+const chalk = require('chalk');
+const ServerlessError = require('../../../../../../classes/Error');
+
+class AwsCompileCloudFrontEvents {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+    this.hooks = {
+      'before:package:compileFunctions': this.prepareFunctions.bind(this),
+      'package:compileEvents': this.compileCloudFrontEvents.bind(this),
+    };
+    if (this.serverless.processedInput.commands[0] === 'remove') {
+      const message = [
+        "Don't forget to manually remove your Lambda@Edge functions ",
+        'once the CloudFront distribution removal is successfully propagated!',
+      ].join('');
+      this.serverless.cli.log(message, 'Serverless', { color: 'orange' });
+    }
+  }
+
+  prepareFunctions() {
+    // Lambda@Edge functions need to be versioned,
+    // can have a max memory size of 128 and a timeout of 5
+    this.serverless.service.getAllFunctions().forEach(functionName => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      if (functionObj.events.find(event => event.cloudFront)) {
+        // ensure that functions are versioned
+        Object.assign(functionObj, { versionFunction: true });
+        // set the maximum memory size if not explicitly configured
+        if (!functionObj.memorySize) {
```

Comment:
I guess it'll be nice to introduce a user friendly validation here. So we crash with meaningful message (before deployment) stating that configured `memorySize` or `timeout` is not valid for CF endpoint case

Reply:
Sounds like a good idea... Just pushed an updated version.